sklearn/linear_model/_coordinate_descent.py

```diff
@@ -1153,8 +1153,43 @@ def fit(self, X, y):
         y : array-like of shape (n_samples,) or (n_samples, n_targets)
             Target values
         """
-        y = check_array(y, copy=False, dtype=[np.float64, np.float32],
-                        ensure_2d=False)
+        # This makes sure that there is no duplication in memory.
```

Comment:
Diff is big here but I mostly just moved the validation of X here, so that we can validate X and y at the same time.